src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -112,7 +112,18 @@ public async Task<ConnectNewHeadersResult> ConsensusTipChangedAsync()
                     this.BestReceivedTip = result.Consumed;
                     this.UpdateBestSentHeader(this.BestReceivedTip);
 
-                    int consumedCount = this.cachedHeaders.IndexOf(result.Consumed.Header) + 1;
+                    uint256 consumedHeaderHash = result.Consumed.Header.GetHash();
```

Comment:
Is this change necessary?

We probably also need to make this in the ProcessesHeadrs method

Reply:
of course it is because it was comparing two different instances of BlockHeader and it returned -1 even if the header was in the list, so I had to compare their hash

Reply:
I could have used FindIndex Linq but I wanted a backward loop so I did an explicit loop